packages/cf-component-icon/src/Icon.js

```diff
@@ -1,33 +1,245 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-class Icon extends React.Component {
-  render() {
-    let className = `cf-icon cf-icon--${this.props.type}`;
+import { createComponent } from '../../cf-style-container';
+
+const Wrapper = createComponent(({
+  color = '#333333',
+  fontSize = 'initial',
+  border = false,
+  spin = false,
+  type,
+  theme
+}) => {
+  const additionalRules = {};
+  let fontPoint = '';
+
+  switch (type) {
```

Comment:
I kind of want this to error if there is no type. I think I might either do a check in the very beginning to `type` and throw an error there or in the `default` case. Left it open since I wasn't sure at the time of writing, will change this soon.